### PR TITLE
Add more `package:test` (removing usages of `package:litetest`)

### PR DESCRIPTION
**Pull request by matanlurey**
... also removed a few TODOs, deleted a proof of concept directory.

Work towards https://github.com/flutter/flutter/issues/133569.

### CODE REVIEW

File: testing/run_tests.py [hunk 1]
```diff
@@ -919,8 +922,7 @@ def gather_dart_package_tests(build_dir, package_path, extra_opts):
     #
     # Until then, assert that no extra_opts are passed and explain the limitation.
     assert not extra_opts, '%s uses package:test and cannot use CLI args' % package_path
-    # TODO(https://github.com/flutter/flutter/issues/154263): Restore `--disable-dart-dev`.
-    opts = ['test']
```
Comment (loic-sharma) [inline testing/run_tests.py:923]: Did you intend to add `--disable-dart-dev` since you removed the TODO and the bug has been fixed?

**Review by loic-sharma — APPROVED** [review]
LGTM

File: testing/run_tests.py (hunk 1 above)
Comment (matanlurey) [reply testing/run_tests.py:923]: No apologies, I should have clarified - it turns out that flag is no longer needed:
https://github.com/flutter/flutter/issues/154268

**Review by jtmcdole — APPROVED** [review]

File: testing/run_tests.py (hunk 1 above)
Comment (loic-sharma) [reply testing/run_tests.py:923]: Gotcha, thanks for the link

**Comment by gaaclarke** [comment]
I didn't see this PR and started migrating one package myself.  I think you are running into the same problem I ran into here: https://github.com/flutter/engine/pull/54909

When invoking litetest it looks like we didn't have asserts turned on so tests would `expect` that we'd get a StateError, but instead we'd get an assertion error when executing with `dart test`.  I solved it by removing the assertion in the skia gold client since the `StateError` is thrown in the same exact case after the assert.

Example failure:
```
00:00 [32m+12[0m[31m -1[0m: test/skia_gold_client_test.dart: fails if GOLDCTL is not set [1m[31m[E][0m[0m
  'package:skia_gold_client/skia_gold_client.dart': Failed assertion: line 193 pos 7: 'isAvailable(environment: _environment)': Trying to use `goldctl` in an environment where it is not available
  dart:core                                             _AssertionError._throwNew
  package:skia_gold_client/skia_gold_client.dart 193:7  SkiaGoldClient._goldctl
  package:skia_gold_client/skia_gold_client.dart 212:7  SkiaGoldClient.auth
  test/skia_gold_client_test.dart 85:22                 main.<fn>
  ```

**Pull request merged**